### DISCUSSION TITLE
Should "Disable rounded corners in Windows 11" work in 24H2?

### DISCUSSION THREAD

**Discussion by muellerto** [ramensoftware/windhawk#340] [General]
In my case it does nothing. Also "Bring back the borders". (But I hate round corners, I get eye cancer from this ...)

I got interested in Windhawk because ExplorerPatcher now also doesn't handle round corners correctly anymore on 24H2.

**Comment by m417z** [comment]
Just tried "Disable rounded corners in Windows 11" on my 24H2 and it seems to be working. Note that you need to add `dwm.exe` to Windhawk's inclusion list, check out the mod description for details.

**Reply by muellerto** [reply]
I did this. I wrote "dwm.exe" into this list. Do I have to qualify this with the full path or something similar?

OK, when you say it works this will be a good reason for my to try it tomorrow again on my Windows machine.

**Reply by m417z** [reply]
You can use the full path, but just `dwm.exe` is enough, see [the screenshot in the mod description](https://windhawk.net/mods/disable-rounded-corners) for reference.

**Comment by muellerto** [comment]
No. In my case corners are always rounded, whatever I include there. I even switched off the exclude of critical system processes for an experiment, doesn't matter at all.

The mod is also not listed among the loaded mods. There's no entry. I have 6 mods activated, 5 are listed.

My OS is a **Windows 11 Pro, 24H2, build 26100.2454**. This info is from my Settings | System | Information.

Surprising and rather strange is that the OS shown in the bottom right corner of my screen (I switched this on) tells me **Windows 11 Pro Insider Preview Build 26100.ge_release.240331-1435** - I never installed an insider preview in my lifetime! I got this with the last ordinary update.

Two days ago I got the big update from 23H2 to 24H2, and then an hour later I got the first cumulative update for 24H2 (KB5046740).

My dwm.exe has actually a size of 110592 bytes, file version in the Properties dialog is 10.0.26100.2454.

**Reply by m417z** [reply]
Perhaps that's related to the insider build you're using.

Please attach debug logs by following the instructions on the Troubleshooting page:
https://github.com/ramensoftware/windhawk/wiki/Troubleshooting#some-or-all-windhawk-mods-dont-work

> * Disable the mod.
> * Go to the mod's Advanced tab.
> * Set **Debug logging** to **Detailed debug logs**.
> * Click on **Show log output**.
> * Enable the mod.
> * Verify that the mod doesn't work, copy the log and save it to a file.
> * Set **Debug logging** back to **None**.

**Comment by muellerto** [comment]
This shows nothing. Not a single line. When I let other mods write a log I get something. This mod "Disable rounded corners" seems to fail very early.

Note the [special approach](https://github.com/valinet/ep_dwm/tree/f56d477c50cc87d596c02f7888c5746ee672d6eb) that ExplorerPatcher made in the last days, which also fails now. The last Windows update brought some breaking changes.

**Reply by m417z** [reply]
It means that Windhawk didn't inject the code successfully. One guess is that perhaps you're using the portable version of Windhawk and you aren't running it as administrator.

**Reply by muellerto** [reply]
Ha! When I start windhawk.exe as Administrator it works immediately. Good to find this out.

But my wish is to start it from an entry in `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Run` or at least from a batch file started at user logon. OK, I could use the [elevate tool](https://www.winability.com/elevate/) for this but a bad thing is that the start as Administrator always leads to a damned full screen MessageBox which must be answered.

**Reply by m417z** [reply]
Is there a reason to use the portable version? I suggest to use the standard version which takes care of running on startup and elevation.

**Reply by muellerto** [reply]
The reason is that I didn't notice the functional difference between these versions ... OK, I can change that.

**Reply by muellerto** [reply]
Indeed I used the releases section here in GitHub (as I do in many other cases). The latest release contains `windhawk_setup.exe` and `windhawk_setup_offline.exe`. I can't see here what is portable and what's not, so I took the smaller one.

**Reply by m417z** [reply]
Both can be used to either install the standard version or to extract the portable version. It's a matter of choosing "Standard" during the installation.

**Reply by muellerto** [reply]
Ah, got it, this "portable" is a question later ... sorry.

**Reply by muellerto** [reply]
Now I understand that. I used the portable installation because the dialog says that the standard installation is for all users. I don't want that because this is a machine of my employer and not a private one. So I installed it only for me never thinking about that this could be the reason for such problems.

When I use the standard installation do then all users get a running windhawk process with the hawk in the system tray or even an open window after login?

**Reply by m417z** [reply]
Unfortunately, Windhawk doesn't truly support per-user customization at this time, see https://github.com/ramensoftware/windhawk/issues/326.

> When I use the standard installation do then all users get a running windhawk process with the hawk in the system tray or even an open window after login?

Yes, all users get Windhawk running, and a tray icon. The tray icon can be hidden for all users in the settings. The main window doesn't open on startup.
